types/transaction.go

```diff
@@ -186,6 +188,9 @@ type Transaction interface {
 	// MatchedRules returns the rules that have matched the requests with associated information.
 	MatchedRules() []MatchedRule
 
+	// DebugLogger returns the debug logger for this transaction.
+	DebugLogger() debuglogger.Logger
```

Comment:
Adding this allows us to centralize logs in connectors and also be in sync with the changes made through WAF directives (otherwise if we pass a logger and the output or the level is changed inside waf due to immutability we don't keep track of that in the provided log).